designs/cloudWatch/cloudWatchIntegration.md

```diff
@@ -0,0 +1,77 @@
+
+## Comparison to other products
+
+### AWS Toolkit for VSCode  
+Not supported
+
+### AWS Toolkit for Visual Studio  
+Supports listing of Lambda CloudWatch Log Streams and downloading them to a temporary file and opening it in Notepad
+
+## Experience
+
+### View Log Groups
+The AWS Explorer gains a CloudWatch root node for interacting wit CloudWatch service. The node lists all the Log Groups that exist in the selected region for the selected credential profile as child nodes. Double clicking a Log Group opens the CloudWatch tool window with the tab for the Log Streams that belong to that group. A new tab is in the tool window is created if one does not already exist.
+
+*Browsing Log Groups in AWS Explorer:*<br />
+![BROWSE_LOG_GROUP]
+
+#### Alternatives Considered
+Instead of building a new CloudWatch service node in the AWS Explorer, we instead build a CloudWatch Log Group browser tab in the CloudWatch tool window. This new tab is also be linked to the active AWS connection settings. This differs from the other tabs which may lead to a confusing user experience and seen as a con.
+
+*Possible alternative for browsing Log Groups:*<br />
+![ALT_BROWSE_LOG_GROUP]
+
+### View Log Streams
+Upon selecting a Log Group, a new tab in the CloudWatch tool window is shown to the user. This new tab lists all the streams for the requested group to allow the user to find the exact stream they are looking for.
+
+Log Streams can also be sorted by name or last event time in order to help facilitate finding the Log Stream of interest.
+
+*View all Log Streams in the selected Log Group:*<br />
+![BROWSE_LOG_STREAMS]
+
+#### Navigate to a Log Stream
+Double clicking the Log group in the AWS Explorer opens the Log Stream selection. 
+
+For additional service integrations, such as a Lambda function, where we know what Log Group is associated with the resource a new context menu action adds the ability to bypass having to search for the Log Group and instead open the Log Stream selection directly.
+
+### View A Log Stream
+Each Log Stream the user selects opens in its own tab to allow for browsing multiple streams at the time. The Log Stream tab is scoped to the region and credentials used to open it so that the user can browse across regions and accounts at the same time.
+
+*View of a Log Stream:*<br />
+![VIEW_LOG_STREAM]
+
+#### Treat as...
+The Treat As feature acts as a Log Event post-processor to convert the Log Event into a more readable format. This feature provides a pre-defined list of formats such as JSON, Plain Text, CSV, or [CLF](https://httpd.apache.org/docs/1.3/logs.html#common). The formats process each Log Event and convert them into a more tabular format where each column maps to an extract format. The filter feature filters on these extracted values.
```

Comment:
This seems like it would be a supplemental feature rather than a p0 feature. Would you agree?

IMO, the IDE seems like a quick and convenient way to view logs. But for more powerful features, the user might turn elsewhere. (e.g. Log services like Graylog, Splunk, etc.)

Reply:
I disagree, the IDE should improve the DevX as much as it can in order to ease development. We shouldn't force customers to onboard to another external system for this, especially since our own services can and do produce logs in this format.

Reply:
Sure, we're in agreement there. 

The question is the cost to implement *and maintain* these features throughout the life-cycle of the plugin and the future evolution of CloudWatch as a service (as well as components like the CloudWatch Console / Insights).

That needs to be weighted against the opportunity cost of not implementing / maintaining other features.

Reply:
My hope is this feature is cheap and can be accomplished with a pluggable interface `fun format(str: String): List<Pair<String, String>> (Column, Value)`, so that the ROI is high and cost to implement is low. We can always revisit it during implementation and cut it then, right?.